### ISSUE TITLE
PedanticGuard and AllOrNothingGuard cannot handle ZST

### ISSUE THREAD

**Issue by lizhuohua** [nabijaczleweli/safe-transmute-rs#65]
Hi, I found that the `check` method for `PedanticGuard` and `AllOrNothingGuard` would panic when `T` is a Zero-Sized Type (ZST).
This is caused by the remainder operation `bytes.len() % size_of::<T>()` in the source code. For example:
```rust
use safe_transmute::guard::AllOrNothingGuard;
use safe_transmute::guard::Guard;

struct Zst;  // Zero-Sized Type

fn main() {
    // Panic: attempt to calculate the remainder with a divisor of zero
    println!("{:?}", AllOrNothingGuard::check::<Zst>(&[0x00, 0x01]));
}
```
Although transmuting something into a ZST might be meaningless, since `Guard` is used for sanity check, probably it is better to handle this instead of causing a panic.

**Comment by nabijaczleweli** [comment]
Can you try with current `master` branch (at least bd64959fbf87a9e05f212287d7bb8b91a9182fab)?

**Comment by lizhuohua** [comment]
Now it works for me! Thanks for your great work!

**Comment by nabijaczleweli** [comment]
Released in [`v0.11.1`](https://github.com/nabijaczleweli/safe-transmute-rs/releases/tag/v0.11.1).
